R/ExtractData_Soils.R

```diff
@@ -263,6 +263,212 @@ extract_soil_CONUSSOIL <- function(MMC, sim_size, sim_space, dir_ex_soil,
+          ils <- seq_len(layer_N)
+          g <- raster::brick(tif_file)
+          soil_data <- raster::crs(g)
+
+          if (!raster::compareCRS(sim_space[["crs_sites"]], soil_data)) {
+            #transform graphics::points to grid-coords
+            sites_conus <- sp::spTransform(sites_conus, CRS = soil_data)
+          }
+          if (sim_space[["scorp"]] == "point") {
+            cell_res_conus <- NULL
+            args_extract <- list(y = sites_conus, type = sim_space[["scorp"]])
+          } else if (sim_space[["scorp"]] == "cell") {
+            cell_res_conus <- align_with_target_res(
+                                        res_from = sim_space[["sim_res"]],
+                                        crs_from = sim_space[["sim_crs"]],
+                                        sp = sim_space[["run_sites"]][todos, ],
+                                        crs_sp = sim_space[["crs_sites"]],
+                                        crs_to = soil_data)
+            args_extract <- list(y = cell_res_conus, coords = sites_conus,
+                                 method = "block", type = sim_space[["scorp"]])
+          }
+          # actually extract the sand values from the sites
+          cond30 <- compiler::cmpfun(function(v) ifelse(is.na(v) | v < 30, NA,
+                                                        v))
+          ftemp <- tif_file
+          g <- if (file.exists(ftemp)) {
+            raster::brick(ftemp)
+          } else {
+            raster::calc(g, fun = cond30, filename = ftemp)
+          }
+          # get max depths for site soil profiles
+          ftemp_d <- paste0(dir.ex.gridded, "/depth_M_250m.tif");
+          gd <- if (file.exists(ftemp_d)) {
+            raster::brick(ftemp_d)
+          } else {
+            raster::calc(gd, fun = cond30, filename = ftemp_d)
+          }
+          soil_frame_depth <- do.call("extract_rSFSW2", args = c(args_extract,
+                                                                 x = list(gd)))
+          MMC[["data"]][todos, grep("depth", MMC[["cn"]])] <- soil_frame_depth
+          if (file_type != "depth"){
+            # get soil data as a dataframe
+            soil_frame <- do.call("extract_rSFSW2", args = c(args_extract,
+                                                             x = list(g)))
+            percent_div <- 100
+            # write density data, MMC[["data"]] requires a different column
+            # name to be written then the rest
+            if (file_type == "matricd"){
+              percent_div <- 1000;
+              MMC[["data"]][todos, grep("density", MMC[["cn"]])[ils]][,
+                                        soil_layer] <- soil_frame / percent_div
+            }
+            else if (file_type == "GravelContent"){
+              soil_frame <- pmax(pmin(soil_frame / 100, 1), 0)
+              # write gravel data to "data
+              # gravel is already in form we want so no need to divide
```

Comment:
Well, line 431 just divided it by hard-coded `100` making the intention of lines 422 and 426 even more confusing.